### ISSUE TITLE
Represent Investigation world coordinates of all linked Datasets

### ISSUE THREAD

**Issue by annehaley** [atlascope/atlascope#77]
The Investigation model should include a new JSONField to contain a mapping of datasets to world coordinates. 
For every dataset linked in `investigation.datasets`, there should be a corresponding key in this JSON. 

See https://github.com/atlascope/atlascope/blob/geojs-projections/web/src/datasets.ts for how @mcovalt describes spatial relationships in the demo code. In this example, datasets are given spatial coordinates relative to each other, but it may be slimmer to express the location of a dataset relative to the "world" of the Investigation. That way, there would be less redundancy, and a dataset could exist in the Investigation while unconnected to all other datasets.

**Comment by waxlamp** [comment]
@mcovalt can you give some input as to what sort of types or structure you need from this?

I was thinking this would be a somewhat special purpose expression of *just* the "embedded" datasets, organized into a tree. But instead it might make sense to simply have a table for these that describes the location and scale and embedding dataset for each embedded dataset.

**Comment by mcovalt** [comment]
While we could represent this data as a JSON, I think it is best to use foreign keys since this describes relationships. See below for a possible (untested) Django model:

```python
from django.contrib.gis.db import models

class RelatedDatasetLocation(models.Model):
    investigation = models.ForeignKey('core.Investigation')
    from_dataset = models.ForeignKey('core.Dataset')
    to_dataset = models.ForeignKey('core.Dataset', related_name='+')
    location = models.PointField()
```

This is an [adjacency list](https://en.wikipedia.org/wiki/Adjacency_list). I think constraints could be added to prevent cycles, but I'd have to think about that for a while.

Getting this list per investigation is something like this: `Investigation.objects.relateddatasetlocation_set()`. We could build the tree from that list in Python of Javascript. I'd probably prefer to use Javascript.

**Comment by waxlamp** [comment]
I believe this was closed by #81; please re-open if not.
